### PR TITLE
Use `[[maybe_unused]]` instead of `(void)` cast

### PR DESCRIPTION
**Pull request by ChrisPec27**
Removed last four void casts in source/distributed/tria.cc and replaced with [[maybe_unused]]. 
Working towards fixing issue #17247

### CODE REVIEW

**Review by tamiko — APPROVED** [review]

File: source/distributed/tria.cc [hunk 1]
```diff
@@ -4285,10 +4286,10 @@ namespace parallel
     template <int dim, int spacedim>
     TemporarilyMatchRefineFlags<dim, spacedim>::TemporarilyMatchRefineFlags(
       dealii::Triangulation<dim, spacedim> &tria)
-      : distributed_tria(
-          dynamic_cast<
-            dealii::parallel::distributed::Triangulation<dim, spacedim> *>(
-            &tria))
+      :
+      [[maybe_unused]] distributed_tria(
```
Comment (tamiko) [inline source/distributed/tria.cc:4290]: You cannot add the attribute to a class member initialization in the constructor.

**Review by tamiko — CHANGES_REQUESTED** [review]

File: source/distributed/tria.cc (hunk 1 above)
Comment (ChrisPec27) [reply source/distributed/tria.cc]: The attribute has been removed from class member initialization in the constructor.

**Review by bangerth — COMMENTED** [review]
Nice. Do you know how to "squash" two commits into one? If not, ask one of us tomorrow!

**Review by tamiko — APPROVED** [review]

**Comment by tamiko** [comment]
I fear that we hit a compiler bug with this:
```/home/runner/work/dealii/dealii/source/distributed/tria.cc:3205:1: error: ‘maybe_unused’ attribute directive ignored [-Werror=attributes]
 3205 |       find_point_owner_rank(
      | ^     ~~~~~~~~~~~~~~~~~~~~~~ 
```
The warning is bogus.

**Review by gassmoeller — APPROVED** [review]
With #17524 we can now merge these PRs and then fix the CI for the broken gcc versions.

**Pull request merged**